clusterloader2/state/tuningsets.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package state
+
+import (
+	"fmt"
+	"sync"
+
+	"k8s.io/perf-tests/clusterloader2/api"
+)
+
+// tuningSets represents set of available tuning sets.
+type tuningSets struct {
+	tuningSets map[string]*api.TuningSet
+	lock       sync.RWMutex
+}
+
+func (ts *tuningSets) getTuningSet(name string) (*api.TuningSet, error) {
+	ts.lock.RLock()
+	defer ts.lock.RUnlock()
+
+	value, exists := ts.tuningSets[name]
+	if !exists {
+		return nil, fmt.Errorf("%v TuningSet not found", name)
+	}
+
+	return value, nil
+}
+
+func (ts *tuningSets) setTuningSet(name string, tuningSet *api.TuningSet) error {
+	ts.lock.Lock()
+	defer ts.lock.Unlock()
+
+	_, exists := ts.tuningSets[name]
+	if exists {
+		return fmt.Errorf("%v TuningSet already exists", name)
```

Comment:
We should add this one to validation code actually (though probably a separate PR to avoid extending the one that is already under review).

Reply:
I was thinking of it as as add-only set.
I will add TODO to move this part to validation.